Library/OcDebugLogLib/OcLog.c

```diff
@@ -347,6 +454,58 @@ OcLogAddEntry  (
     }
   }
 
+  return Status;
+}
+
+EFI_STATUS
+EFIAPI
+OcLogAddEntry (
+  IN OC_LOG_PROTOCOL    *OcLog,
+  IN UINTN              ErrorLevel,
+  IN CONST CHAR8        *FormatString,
+  IN VA_LIST            Marker
+  )
+{
+  EFI_STATUS                  Status;
+
+  OC_LOG_PRIVATE_DATA         *Private;
+
+  CHAR8                       *FormatCopy;
+  UINTN                       Index;
+  BOOLEAN                     IsFiltered;
+  OC_LOG_FILTER_MODE          FilterMode;
+
+  ASSERT (OcLog != NULL);
+  ASSERT (FormatString != NULL);
+
+  Private = OC_LOG_PRIVATE_DATA_FROM_OC_LOG_THIS (OcLog);
+
+  if ((OcLog->Options & OC_LOG_ENABLE) == 0) {
+    //
+    // Silently ignore when disabled.
+    //
+    return EFI_SUCCESS;
+  }
+
+  //
+  // Filter log.
+  //
+  FormatCopy = AllocateCopyPool (AsciiStrSize (FormatString), FormatString);
+  ASSERT (FormatCopy != NULL);
+  if (Private->FlexFilters != NULL) {
+    for (Index = 0; Index < Private->FlexFilters->Count; ++Index) {
+      IsFiltered = IsPrefixFiltered (FormatString, Private->FlexFilters, &FilterMode);
+      //
+      // Drop the filter on unknown mode.
+      //
+      if (!IsFiltered || (IsFiltered && FilterMode == OcLogFilterModePositive)) {
```

Comment:
The ``IsFiltered`` in the second condition is implicite from the first condition failing

Reply:
https://github.com/acidanthera/OpenCorePkg/commit/f734c64820546ad116ffc39f21b1e2a40f654ddb